legend-depot-core-services/src/main/java/org/finos/legend/depot/services/projects/DependenciesCache.java

```diff
@@ -77,15 +78,28 @@ private Function3<String,String,String,StoreProjectVersionData> getProjectDataFr
 
     private Set<ProjectVersion> calculateTransitiveDependencies(ProjectVersion projectVersion, Map<String, StoreProjectVersionData> projectDataMap)
     {
-        return getTransitiveDependencies(projectVersion, getProjectDataFromProjectsMap(projectDataMap));
+        return getTransitiveDependencies(projectVersion, getProjectDataFromProjectsMap(projectDataMap), false);
     }
 
-    private Set<ProjectVersion> getTransitiveDependencies(ProjectVersion pv, Function3<String,String,String,StoreProjectVersionData> projectDataProvider)
+    private Set<ProjectVersion> getTransitiveDependencies(ProjectVersion pv, Function3<String,String,String,StoreProjectVersionData> projectDataProvider, boolean checkDep)
     {
         Set<ProjectVersion> dependencies = new HashSet<>();
         try
         {
-            StoreProjectVersionData projectData = projectDataProvider.value(pv.getGroupId(), pv.getArtifactId(), pv.getVersionId());
+            StoreProjectVersionData projectData = null;
+            if (checkDep)
+            {
+                Optional<StoreProjectVersionData> versionData = projectsVersionsStore.find(pv.getGroupId(), pv.getArtifactId(), pv.getVersionId());
+                if (!versionData.isPresent())
+                {
+                    String errorMsg = String.format("Could not find dependency %s-%s-%s in store", pv.getGroupId(), pv.getArtifactId(), pv.getVersionId());
+                    LOGGER.error(errorMsg);
+                    throw new RuntimeException(errorMsg);
+                }
+                //TODO: would be better of we update the entry directly in the projectDataMap
+                projectData = versionData.get();
+            }
+            projectData = projectData == null ? projectDataProvider.value(pv.getGroupId(), pv.getArtifactId(), pv.getVersionId()) : projectData;
             if (projectData != null)
```

Comment:
as above if project data is null? wjhy would it be null?

Reply:
when we have a new project version this might not exist in the map and hence the provider could still be null